app/services/messages/reader.rb

```diff
@@ -46,5 +49,13 @@ def visit_based_available?
     def expiration
       message.expiration
     end
+
+    def enqueue_visibility_job?
+      visibility_time.present? && resolver_name.present?
+    end
+
+    def limit_visibility_time
+      DiscordMessages::VisibilityJob.set(wait: visibility_time).perform_later(message.id, resolver_name)
+    end
```

Comment:
Same here. Please, whenever you need to add discord specific calls in `app/services/` let's find a wokaround to avoid it. This can be solved with an event too, something like `privy_message_read` and listen to it on `app/integraions/discord`.

Also keep in mind that any stuff we do in `app/services/` for discord will very likely need to be done for any new provider we want to add so leaving these dependencies here will become very dirty at that point